### PR TITLE
prepare: exit when passed incorrect parameters

### PR DESCRIPTION
**Pull request by Yuggupta27**
Now script does not continue execution
when an invalid argument is passed.

Signed-off-by: Yug Gupta <ygupta@redhat.com>

### CODE REVIEW

File: prepare.sh [hunk 1]
```diff
@@ -49,6 +49,7 @@ while true; do
         ;;
     --)
         shift
+        exit 0
```
Comment (nixpanic) [inline prepare.sh:52]: is this hit when parameters are passed that are not recognized? If so, please display an error and `exit 1`

File: make-containerized-build.groovy [hunk 2]
```diff
@@ -28,7 +28,7 @@ node('cico-workspace') {
 	try {
 		stage('prepare bare-metal machine') {
 			sh 'scp -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no ./prepare.sh root@${CICO_NODE}:'
-			sh 'ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@${CICO_NODE} ./prepare.sh --workdir=/opt/build --gitrepo=${GIT_REPO} --branch=${GIT_BRANCH}'
+			sh 'ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@${CICO_NODE} ./prepare.sh --workdir=tip/ --gitrepo=${GIT_REPO} --ref=pull/${ghprbPullId}/head'
```
Comment (nixpanic) [inline make-containerized-build.groovy:31]: use `/opt/build` as directory (or better, `/opt/build/go/src/github.com/ceph/ceph-csi`), otherwise it lands in some local place (probably `/root/tip`). If we want to use standard go tools, the directory needs to be in the `GOPATH` format.

Comment (nixpanic) [reply make-containerized-build.groovy:31]: There needs to be a check that `ghprbPullId` is available as environment  variable, if it is not, fall back to `--ref=master`.
Time to learn some Groovy :disappointed:

File: prepare.sh [hunk 3]
```diff
@@ -31,7 +31,7 @@ while true; do
         echo "--gitrepo                 specify the git repository" 
         echo " "
         echo "Sample Usage:"
-        echo "./prepare.sh --gitrepo=https://github.com/example --workdir=/opt/build --ref=pull/123/head"
+        echo "./prepare.sh --gitrepo=https://github.com/example --workdir=tip/ --ref=pull/123/head"
```
Comment (nixpanic) [inline prepare.sh:34]: this can be a separate commit. One for the corrections to the prepare.sh script, one for the update to the groovy job

Comment (Yuggupta27) [reply prepare.sh:34]: Thanks, I'll make the required changes

**Comment by Yuggupta27** [comment]
Done!
